src/views/ProductView.vue

```diff
@@ -1,20 +1,77 @@
 <script setup lang="ts">
-  import { useProduct } from "@/composables/api"
   import { watchEffect } from "vue"
   import { useRouter } from "vue-router"
+  import { useProduct } from "@/composables/api"
+  import { capitalize } from "@/utils"
+  import BreadCrumbs from "@/components/product-page/BreadCrumbs.vue"
+  import ImageCarousel from "@/components/product-page/ImageCarousel.vue"
+  import PrimaryButton from "@/components/PrimaryButton.vue"
   import PageTitle from "@/components/PageTitle.vue"
 
   const { product, isProductLoading } = useProduct()
   const router = useRouter()
 
   watchEffect(() => {
     if (isProductLoading.value) return
-    if (!product.value) router.push("/not-found")
+    if (!product.value) router.push({ name: "not-found" })
```

Comment:
Try to get redirected to 404 and then refresh the page. In theory you would expect to stay on the same page